### PR TITLE
Speed up MergeRuns for detector scan workspaces

### LINKED ISSUE #21452: MergeRuns speed up for detector scan workspaces
Improve performance for MergeRuns detector scan cases.

### PR DESCRIPTION
**Pull request by ianbush**
Speeds up `MergeRuns` for detector scan workspaces by parallelising a slow loop.

**To test:**

Run performance test `ctest -R MergeRunsPerformance`. Then delete line 684 in `MergeRuns.cpp` and run again. With the parallelisation there should be a ~20% speed up.

Fixes #21452.

**Release Notes** 
*Does not need to be in the release notes.* - non functional change.

---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####

- [ ] Is the code of an acceptable quality?
- [ ] Does the code conform to the [coding standards](http://www.mantidproject.org/Coding_Standards)?
- [ ] Are the unit tests small and test the class in isolation?
- [ ] If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- [ ] Do changes function as described? Add comments below that describe the tests performed?
- [ ] Do the changes handle unexpected situations, e.g. bad input?
- [ ] Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

File: Framework/Algorithms/test/MergeRunsTest.h [hunk 1]
```diff
@@ -1732,4 +1732,37 @@ class MergeRunsTest : public CxxTest::TestSuite {
...
+      std::string wsName = "a" + std::to_string(i);
+      AnalysisDataService::Instance().addOrReplace(wsName, ws);
+    }
+
+    Mantid::Algorithms::MergeRuns mergeRuns;
+    mergeRuns.initialize();
+    mergeRuns.setPropertyValue("InputWorkspaces",
+                               "a0, a1, a2, a3, a4, a5, a6, a7, a8, a9");
+    mergeRuns.setPropertyValue("OutputWorkspace", "outputWS");
+    mergeRuns.execute();
```
Comment (gvardany) [inline Framework/Algorithms/test/MergeRunsTest.h]: Just a comment, shouldn't only `execute()` be in the test body, the rest in setup/teardown?

Comment (ianbush) [reply Framework/Algorithms/test/MergeRunsTest.h]: Yeah, it probably should, it is a smaller amount of time to do the setup but not insignificant.

**Comment by gvardany** [comment]
Can't test the parallel loop on my system. Leaving for someone who can.

**Review by samueljackson92 — APPROVED** [review]
I'm not quite seeing ~20% faster but I do see a small speed up between tests on osx.

**Pull request merged**